### ISSUE TITLE
Column selection ignored when using accessible_by

### ISSUE THREAD

**Issue by kbrandwijk** [CanCanCommunity/cancancan#521]
### Steps to reproduce

1. Create a model
2. Define an ability with any `:read` configuration for that model
3. Perform a query with a `select()` to specify the columns

Effectively, and I think that's the import part, there's just no way to limit the columns that `cancancan` selects. It always selects all columns for all tables.

### Expected behavior
The generated query should only contain the columns that are part of the select and needed for the ability.

### Actual behavior
The generated query contains all columns for the model. As you can see from the output below, my column selection is included in the query first, but followed by every single column of the model. For a huge model this has quite a big impact on the system.

```
> MyModel.select('mycolumn').all
MyModel Load (0.4ms)  SELECT  "my_models"."mycolumn" FROM "my_models"

> MyModel.select('mycolumn').accessible_by(ability).all
MyModel Load (1.5ms)  SELECT "my_models"."mycolumn", "my_models"."first_column" as t0_r0, "my_models"."second_column" as t0_r1 (clipped)
```

### System configuration
**Rails version**:
5.2.0
**Ruby version**:
2.5.1
**CanCanCan version**
2.2.0

**Comment by coorasse** [comment]
This problem has been discussed also [here](https://github.com/CanCanCommunity/cancancan/issues/556#issuecomment-460219397) and it should be fixed in 3.0.0

**Comment by coorasse** [comment]
This issue should be addressed completely in 3.1.0. Please feel free to re-open otherwise.
